File: sdkwork-ai-sdk-swift/Sources/API/AssistantsApi.swift
import Foundation

public class AssistantsApi {
    private let client: HttpClient
    
    public init(client: HttpClient) {
        self.client = client
    }

    /// Get assistant
    public func getAssistant(assistant_id: String) async throws -> AssistantResponse? {
        let response = try await client.get(ApiPaths.aiPath("/assistants/\(assistant_id)"))
        return response as? AssistantResponse
    }

    /// Update assistant
    public func updateAssistant(assistant_id: String, body: AssistantCreateRequest) async throws -> AssistantResponse? {
        let response = try await client.post(ApiPaths.aiPath("/assistants/\(assistant_id)"), body: body)
        return response as? AssistantResponse
    }

    /// Delete assistant
    public func deleteAssistant(assistant_id: String) async throws -> AssistantDeleteResponse? {
        let response = try await client.delete(ApiPaths.aiPath("/assistants/\(assistant_id)"))
        return response as? AssistantDeleteResponse
    }

    /// List assistants
    public func listAssistants(params: [String: Any]? = nil) async throws -> AssistantListResponse? {
        let response = try await client.get(ApiPaths.aiPath("/assistants"), params: params)
        return response as? AssistantListResponse
    }

    /// Create assistant
    public func createAssistant(body: AssistantCreateRequest) async throws -> AssistantResponse? {
        let response = try await client.post(ApiPaths.aiPath("/assistants"), body: body)
        return response as? AssistantResponse
    }
}

File: sdkwork-ai-sdk-kotlin/src/main/kotlin/com/sdkwork/ai/api/AssistantsApi.kt
package com.sdkwork.ai.api

import com.sdkwork.ai.*
import com.sdkwork.ai.http.HttpClient

class AssistantsApi(private val client: HttpClient) {

    /** Get assistant */
    suspend fun getAssistant(assistant_id: String): AssistantResponse? {
        return client.get(ApiPaths.aiPath("/v1/assistants/$assistant_id")) as? AssistantResponse
    }

    /** Update assistant */
    suspend fun updateAssistant(assistant_id: String, body: AssistantCreateRequest): AssistantResponse? {
        return client.post(ApiPaths.aiPath("/v1/assistants/$assistant_id"), body) as? AssistantResponse
    }

    /** Delete assistant */
    suspend fun deleteAssistant(assistant_id: String): AssistantDeleteResponse? {
        return client.delete(ApiPaths.aiPath("/v1/assistants/$assistant_id")) as? AssistantDeleteResponse
    }

    /** List assistants */
    suspend fun listAssistants(params: Map<String, Any>? = null): AssistantListResponse? {
        return client.get(ApiPaths.aiPath("/v1/assistants"), params) as? AssistantListResponse
    }

    /** Create assistant */
    suspend fun createAssistant(body: AssistantCreateRequest): AssistantResponse? {
        return client.post(ApiPaths.aiPath("/v1/assistants"), body) as? AssistantResponse
    }
}
